File: mylibrary/src/main/java/com/xiaolu/mylibrary/load/LoadUtil.java
package com.xiaolu.mylibrary.load;

import android.os.Looper;

import com.xiaolu.mylibrary.load.target.ITarget;

import java.util.List;

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.load
 * @ClassName: LoadUtil
 * @Description: java类作用描述
 * @Author: xiaol
 * @CreateDate: 2020/8/18 11:15
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/8/18 11:15
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
public class LoadUtil {
    public static boolean isMainThread() {
        return Looper.myLooper() == Looper.getMainLooper();
    }

    public static ITarget getTargetContext(Object target, List<ITarget> targetContextList) {
        for (ITarget targetContext : targetContextList) {
            if (targetContext.equals(target)) {
                return targetContext;
            }

        }
        throw new IllegalArgumentException("No TargetContext fit it");
    }
}

File: mylibraryKotlin/src/main/java/com/xiaolu/mylibrarykotlin/load/LoadUtil.kt
package com.xiaolu.mylibrarykotlin.load

import android.os.Looper
import com.xiaolu.mylibrarykotlin.load.target.ITarget

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.load
 * @ClassName: LoadUtil
 * @Description: java类作用描述
 * @Author: xiaol
 * @CreateDate: 2020/8/18 11:15
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/8/18 11:15
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
object LoadUtil {
    val isMainThread: Boolean
        get() = Looper.myLooper() == Looper.getMainLooper()

    fun getTargetContext(target: Any, targetContextList: List<ITarget>): ITarget {
        for (targetContext in targetContextList) {
            if (targetContext == target) {
                return targetContext
            }
        }
        throw IllegalArgumentException("No TargetContext fit it")
    }
}